### PR TITLE
Skopeo should support for BigFilesTemporaryDir (SystemContext)

### PR DESCRIPTION
**Pull request by rhatdan**
Enhancement request: https://github.com/containers/skopeo/issues/805

Also sorted commands and options on skopeo man page and skopeo --help

Originally submitted by  Michel Belleau <michel.belleau@malaiwah.com>
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
Replaces: https://github.com/containers/skopeo/pull/806

@mtrmac @vrothberg @TomSweeneyRedHat PTAL

**Comment by rhatdan** [comment]
@malaiwah PTAL

File: cmd/skopeo/main.go [hunk 1]
```diff
@@ -85,30 +75,47 @@ func createApp() (*cli.App, *globalOptions) {
...
+			Usage:  "require HTTPS and verify certificates when talking to container registries (defaults to true)",
+			Hidden: true,
+			Value:  newOptionalBoolValue(&opts.tlsVerify),
+		},
+		cli.StringFlag{
+			Name:        "tmpdir",
+			Usage:       "directory used to store temporary files",
+			Destination: &opts.tmpDir,
+			Value:       getTempDir(),
+		},
```
Comment (TomSweeneyRedHat) [inline cmd/skopeo/main.go]: Does this flag do much?  If $TMPDIR is set, won't the system already use that?  Should we instead take a value too and use that directory if it's found, and if not, then default to $TMPDIR and then if that's not set, then /var/tmp?

Comment (rhatdan) [reply cmd/skopeo/main.go]: No the system does not currently use TMPDIR for this.  This chenage gets it to use the tmpdir and allows users to specify alternative directories.

**Comment by rhatdan** [comment]
@mtrmac @vrothberg PTAL

File: cmd/skopeo/main.go [hunk 2]
```diff
@@ -44,32 +45,21 @@ func createApp() (*cli.App, *globalOptions) {
 	}
 	app.Usage = "Various operations with container images and container image registries"
 	app.Flags = []cli.Flag{
+		cli.DurationFlag{
```
Comment (mtrmac) [inline cmd/skopeo/main.go]: (Please split unrelated re-orderings into a separate commit in the future.)

File: docs/skopeo.1.md [hunk 3]
```diff
@@ -46,23 +46,25 @@ Most commands refer to container images, using a _transport_`:`_details_ format.
...
 
   **--override-variant** _VARIANT_ Use _VARIANT_ instead of the running architecture variant for choosing images.
 
-  **--command-timeout** _duration_ Timeout for the command execution.
+  **--policy** _path-to-policy_ Path to a policy.json file to use for verifying signatures and deciding whether an image is trusted, overriding the default trust policy file.
 
-  **--help**|**-h** Show help
+  **--registries.d** _dir_ use registry configuration files in _dir_ (e.g. for container signature storage), overriding the default path.
+
+  **--tmpdir:**_dir_ _dir_ used to store temporary files. Defaults to /var/tmp.
```
Comment (mtrmac) [inline docs/skopeo.1.md]:
```suggestion
  **--tmpdir** _dir_ used to store temporary files. Defaults to /var/tmp.
```

File: cmd/skopeo/main.go [hunk 4]
```diff
@@ -85,30 +75,46 @@ func createApp() (*cli.App, *globalOptions) {
...
+		},
+		cli.GenericFlag{
+			Name:   "tls-verify",
+			Usage:  "require HTTPS and verify certificates when talking to container registries (defaults to true)",
+			Hidden: true,
+			Value:  newOptionalBoolValue(&opts.tlsVerify),
+		},
+		cli.StringFlag{
+			Name:        "tmpdir",
+			Usage:       "directory used to store temporary files",
```
Comment (mtrmac) [inline cmd/skopeo/main.go]: `s/temporary/big temporary/`? (Both here and in the man page.)

Comment (rhatdan) [reply cmd/skopeo/main.go]: That means nothing to the user.

Comment (mtrmac) [reply cmd/skopeo/main.go]: OTOH the current version is just incorrect. The difference can be explained in the man page for those who care.

\*shrug\* if someone else wants to merge this as is, feel free.

**Pull request merged**